sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StreamingInserts.java

```diff
@@ -38,20 +38,39 @@
   private InsertRetryPolicy retryPolicy;
 
   /** Constructor. */
-  StreamingInserts(CreateDisposition createDisposition,
+  public StreamingInserts(CreateDisposition createDisposition,
                    DynamicDestinations<?, DestinationT> dynamicDestinations) {
+    this(createDisposition, dynamicDestinations, InsertRetryPolicy.alwaysRetry());
+  }
+
+  /** Constructor. */
+  public StreamingInserts(CreateDisposition createDisposition,
+                          DynamicDestinations<?, DestinationT> dynamicDestinations,
+                          InsertRetryPolicy retryPolicy) {
     this.createDisposition = createDisposition;
     this.dynamicDestinations = dynamicDestinations;
     this.bigQueryServices = new BigQueryServicesImpl();
-    this.retryPolicy = InsertRetryPolicy.alwaysRetry();
+    this.retryPolicy = retryPolicy;
+  }
+
+  /** Constructor. */
+  private StreamingInserts(CreateDisposition createDisposition,
+                          DynamicDestinations<?, DestinationT> dynamicDestinations,
+                          BigQueryServices bigQueryServices,
+                          InsertRetryPolicy retryPolicy) {
+    this.createDisposition = createDisposition;
+    this.dynamicDestinations = dynamicDestinations;
+    this.bigQueryServices = bigQueryServices;
+    this.retryPolicy = retryPolicy;
   }
 
   void setTestServices(BigQueryServices bigQueryServices) {
```

Comment:
Maybe this should be a `with` method too, now that you have such convenient constructors.